src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -177,1327 +205,1182 @@ static std::vector<rct_widget> _window_editor_object_selection_widgets = {
 
-    // Draw object author (will be blank space if no author in file or a non JSON object)
-    {
-        auto ft = Formatter();
-        std::string authorsString;
-        for (size_t i = 0; i < listItem->repositoryItem->Authors.size(); i++)
+        // Draw preview
         {
-            if (i > 0)
+            rct_drawpixelinfo clipDPI;
+            auto screenPos = windowPos + ScreenCoordsXY{ previewWidget.left + 1, previewWidget.top + 1 };
+            _width = previewWidget.width() - 1;
+            height = previewWidget.height() - 1;
+            if (clip_drawpixelinfo(&clipDPI, &dpi, screenPos, _width, height))
             {
-                authorsString.append(", ");
+                _loadedObject->DrawPreview(&clipDPI, _width, height);
             }
-            authorsString.append(listItem->repositoryItem->Authors[i]);
         }
-        ft.Add<rct_string_id>(STR_STRING);
-        ft.Add<const char*>(authorsString.c_str());
-        DrawTextEllipsised(
-            dpi, { w->windowPos.x + w->width - 5, screenPos.y }, w->width - w->widgets[WIDX_LIST].right - 4,
-            STR_WINDOW_COLOUR_2_STRINGID, ft, { TextAlignment::RIGHT });
-    }
-}
-
-/**
- *
- *  rct2: 0x006AAB56
- */
-static void WindowEditorObjectSelectionPaint(rct_window* w, rct_drawpixelinfo* dpi)
-{
-    int32_t width;
 
-    WindowDrawWidgets(w, dpi);
-
-    // Draw tabs
-    for (size_t i = 0; i < std::size(ObjectSelectionPages); i++)
-    {
-        const auto& widget = w->widgets[WIDX_TAB_1 + i];
-        if (widget.type != WindowWidgetType::Empty)
+        // Draw name of object
         {
-            auto image = ImageId(ObjectSelectionPages[i].Image);
-            auto screenPos = w->windowPos + ScreenCoordsXY{ widget.left, widget.top };
-            gfx_draw_sprite(dpi, image, screenPos);
+            auto screenPos = windowPos + ScreenCoordsXY{ previewWidget.midX() + 1, previewWidget.bottom + 3 };
+            _width = this->width - widgets[WIDX_LIST].right - 6;
+            auto ft = Formatter();
+            ft.Add<rct_string_id>(STR_STRING);
+            ft.Add<const char*>(listItem->repositoryItem->Name.c_str());
+            DrawTextEllipsised(&dpi, screenPos, _width, STR_WINDOW_COLOUR_2_STRINGID, ft, { TextAlignment::CENTRE });
         }
-    }
 
-    const int32_t ride_tabs[] = {
-        SPR_TAB_RIDE_16,        IMAGE_TYPE_REMAP | SPR_TAB_RIDES_TRANSPORT_0,
-        SPR_TAB_RIDES_GENTLE_0, IMAGE_TYPE_REMAP | SPR_TAB_RIDES_ROLLER_COASTERS_0,
-        SPR_TAB_RIDES_THRILL_0, SPR_TAB_RIDES_WATER_0,
-        SPR_TAB_RIDES_SHOP_0,   SPR_TAB_FINANCES_RESEARCH_0,
-    };
-    const int32_t ThrillRidesTabAnimationSequence[] = {
-        5, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 4, 0, 0, 0,
-    };
+        //PaintDescriptions
 
-    // Draw ride tabs
-    if (GetSelectedObjectType(w) == ObjectType::Ride)
-    {
-        for (int32_t i = 0; i < 7; i++)
+        const auto& widget = widgets[WIDX_PREVIEW];
```

Comment:
revert this back into two functions.

Reply:
Done https://github.com/OpenRCT2/OpenRCT2/pull/16027/commits/3bb6b5c5cd3993c7705e8fea5e138633ee418021 Just to be sure, was this was about the PaintDescriptions and PaintDebugData functions being in OnDraw?